libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -353,18 +350,18 @@ - (void)enqueueRequest:(SFRestRequest *)request
             // 2xx indicates success.
             if ([SFRestAPI isStatusCodeSuccess:statusCode]) {
                 id dataForDelegate = [strongSelf prepareDataForDelegate:data request:request response:response];
-                if (successBlock) {
-                    successBlock(dataForDelegate, response);
+                if (request.successBlock) {
```

Comment:
This is the main change, use the callback on the individual request, otherwise the callback on one of the requests is used for all the requests

Reply:
Was it a recent regression? Or was it always like that?